salt/states/dockerng.py

```diff
@@ -2145,7 +2145,7 @@ def volume_present(name, driver=None, driver_opts=None):
             return ret
     # volume exits, check if driver is the same.
     volume = volumes[0]
-    if volume['Driver'] != driver:
+    if driver is not None and volume['Driver'] != driver:
```

Comment:
> I think it is better to initiate driver parameter with the default docker's volume driver name.

I disagree—I think it's better to interpret the lack of a specified driver as "I don't care what the driver is" rather than interpret it as "I care very much that the driver matches the Docker default."

(I also think it's a bad idea to delete a volume just because the driver is different.  I think it'd be better to error out than to risk surprising the user with permanent data loss.  But that's best addressed in a different merge request.)

>     driver = get_default_docker_driver()

Does Docker provide a way to get the default driver?  (other than create a temporary volume using defaults, query its properties, then delete it)  If not, I'm not sure how one would implement `get_default_docker_driver()`.  I'd be worried about hard-coding `local` in Salt, as that will cause problems if Docker ever changes the default to something else.

If we can figure out how to cleanly implement `get_default_docker_driver()`, perhaps we can implement a compromise:  `None` would mean "I don't care" while `default` would mean "I care that it matches the Docker default".  (And then hope that Docker never adds a driver named `default` whose semantics differ.)